frontend/src/pages/Login.js

```diff
@@ -0,0 +1,152 @@
+            dispatch(settings.actions.setShortBreakMinutes(data.response.shortBreakMinutes))
+            dispatch(settings.actions.setLongBreakMinutes(data.response.longBreakMinutes))
+            dispatch(user.actions.setError(null))
+          })
+        } else {
+          batch(() => {
+            dispatch(user.actions.setUserId(null))
+            dispatch(user.actions.setUsername(null))
+            dispatch(user.actions.setAccessToken(null))
+            dispatch(settings.actions.setWorkMinutes(25))
+            dispatch(settings.actions.setShortBreakMinutes(5))
+            dispatch(settings.actions.setLongBreakMinutes(15))
+            dispatch(user.actions.setError(data.response))
+          })
+        }
+      })
+  }
+
+  return (
+    <>
+      <LoginContainer>
+        <FormWrapper>
+          <h3 style={{ color: "var(--red)" }}>
+            {mode === "signin"
+              ? "Log in to your account"
+              : "Create new account"}
+          </h3>
+          <form onSubmit={onFormSubmit}>
+            <UserInfoWrapper>
+              <legend>Username:</legend>
+              <UserInput
+                type="text"
+                value={username}
+                onChange={(e) => setUsername(e.target.value)}
+              />
+            </UserInfoWrapper>
+            <UserInfoWrapper>
+              <legend>Password:</legend>
+              <UserInput
+                type="password"
+                value={password}
+                onChange={(e) => setPassword(e.target.value)}
+              />
+            </UserInfoWrapper>
+            {error && (
+              <div>
+                <p
+                  style={{
+                    color: "red",
+                    textAlign: "center",
+                    fontWeight: "bold",
+                  }}
+                >
+                  {error}
+                </p>
+              </div>
+            )}
+            <LoginButton type="submit">
+              {mode === "signup" ? "Submit" : "Log in"}
+            </LoginButton>
+            <div>
+              {mode === "signin" ? (
+                <LinkText>
+                  <p>New to our app? </p>
+                  <p
+                    onClick={() => setMode("signup")}
+                    style={{
+                      fontWeight: "700",
+                      cursor: "pointer",
+                      textDecoration: "underline",
+                    }}
+                  >
+                    Create an account
+                  </p>
+                </LinkText>
+              ) : (
+                <LinkText>
+                  <p>Already have an account? </p>
+                  <p
+                    onClick={() => {
+                      setMode("signin")
+                      dispatch(user.actions.setError(null))
+                      setUsername("")
+                      setPassword("")
+                    }}
```

Comment:
Too much logic to have inside JSX expression - let's move it to separate function